database_wrapper.py

```diff
@@ -247,72 +250,76 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-class NimbusMySQLAlchemy():  # NimbusMySQLAlchemy(NimbusDatabase):
+class NimbusMySQLAlchemy:  # NimbusMySQLAlchemy(NimbusDatabase):
     """
     """
 
-    def __init__(self, config_file: str = 'config.json') -> None:
-        # # sqlalchemy needs a Base class for all the database entities
-        # self.Base = declarative_base()
+    def __init__(self, config_file: str = "config.json") -> None:
         self.engine = None  # gets set according to config_file
         self.Courses = Courses
         self.AudioSampleMetaData = AudioSampleMetaData
+        self.QuestionAnswerPair = QuestionAnswerPair
 
         with open(config_file) as json_data_file:
             config = json.load(json_data_file)
 
-        if config.get('mysql', False):
-            mysql_config = config['mysql']
+        if config.get("mysql", False):
+            mysql_config = config["mysql"]
             RDBMS = "mysql"
             PIP_PACKAGE = "mysqlconnector"
             SQLALCHEMY_DATABASE_URI = "{}+{}://{}:{}@{}:{}/{}".format(
-                RDBMS, PIP_PACKAGE, mysql_config['user'],
-                mysql_config['password'], mysql_config['host'],
-                mysql_config['port'], mysql_config['database'])
+                RDBMS,
+                PIP_PACKAGE,
+                mysql_config["user"],
+                mysql_config["password"],
+                mysql_config["host"],
+                mysql_config["port"],
+                mysql_config["database"],
+            )
             self.engine = create_engine(SQLALCHEMY_DATABASE_URI)
 
             if self.engine is None:
-                raise BadConfigFileError('failed to connect to MySQL')
+                raise BadConfigFileError("failed to connect to MySQL")
         else:
-            msg = "config.json is missing {} field.".format('mysql')
+            msg = "config.json is missing {} field.".format("mysql")
             raise BadConfigFileError(msg)
 
         self.inspector = inspect(self.engine)
         self._create_database_session()
         print("initialized NimbusMySQLAlchemy")
 
     def _create_all_tables(self):
-        # TODO: reconsider if this even works???
-        # self.Base.metadata.create_all(self.engine)
-        # TODO: go with this instead... more explicit...
         def __safe_create(SQLAlchemy_object):
             table_name = SQLAlchemy_object.__tablename__
-            print("creating {}...".format(table_name))
+            print(f"creating {table_name}...")
```

Comment:
I'm not familiar with this syntax - is this how we want to do string formatting?

Reply:
Yes, `f-strings` are a thing https://realpython.com/python-f-strings/

They are super cool!

Thanks for your curiosity @snekiam